plugins/rule_engines/CMakeLists.txt

```diff
@@ -1,82 +1,61 @@
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/index.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/nre.reHelpers1.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/nre.reHelpers2.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/nre.reLib1.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/parser.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/reVariableMap.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/reVariableMap.gen.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/restructs.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/rules.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/typing.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/utils.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/rsRe.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/functions.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/libirods_rule_engine_plugin-irods_rule_language.cpp
-  )
-
-set(
-  IRODS_RULE_ENGINE_PLUGIN-CPP_DEFAULT_POLICY_SOURCES
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-cpp_default_policy/libirods_rule_engine_plugin-cpp_default_policy.cpp
-  )
+cmake_minimum_required(VERSION 3.12.0 FATAL_ERROR)
+# target_link_libraries for object targets
 
-set(
-  IRODS_RULE_ENGINE_PLUGIN-PASSTHROUGH_SOURCES
-  ${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-passthrough/libirods_rule_engine_plugin-passthrough.cpp
-  )
+add_subdirectory(irods_rule_language)
 
 set(
   IRODS_RULE_ENGINE_PLUGINS
-  irods_rule_engine_plugin-irods_rule_language
-  irods_rule_engine_plugin-cpp_default_policy
-  irods_rule_engine_plugin-passthrough
-  )
+  irods_rule_language
+  cpp_default_policy
+  passthrough
+)
+
+foreach(plugin IN LISTS ${IRODS_RULE_ENGINE_PLUGINS})
+  set(plugin_target "irods_rule_engine_plugin-${plugin}")
+  set(plugin_name "${plugin_target}")
 
-foreach(PLUGIN ${IRODS_RULE_ENGINE_PLUGINS})
-  string(TOUPPER ${PLUGIN} PLUGIN_UPPERCASE)
+  if (TARGET ${plugin_target})
+    continue()
+  endif()
 
   add_library(
-    ${PLUGIN}
+    ${plugin_target}
     MODULE
-    ${${PLUGIN_UPPERCASE}_SOURCES}
-    )
-
-  target_include_directories(
-    ${PLUGIN}
-    PRIVATE
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/irods_rule_engine_plugin-irods_rule_language/include>
-    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/server/main_server/include>
-    ${IRODS_EXTERNALS_FULLPATH_BOOST}/include
-    ${IRODS_EXTERNALS_FULLPATH_FMT}/include
-    )
-
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/${plugin}.cpp"
+  )
+  #set_property(TARGET ${plugin_target} PROPERTY LIBRARY_OUTPUT_NAME ${plugin_name})
```

Comment:
> cpp_default_policy REP is required for irods_rule_language to work

i don't... think so?   it's just another implementation of the same policies, just in C++...   but we *do* want our default installation to include both.

Reply:
```bash
$ cat /etc/irods/server_config.json | jq '.plugin_configuration.rule_engines'
[
  {
    "instance_name": "irods_rule_engine_plugin-irods_rule_language-instance",
    "plugin_name": "irods_rule_engine_plugin-irods_rule_language",
    "plugin_specific_configuration": {
      "re_data_variable_mapping_set": [
        "core"
      ],
      "re_function_name_mapping_set": [
        "core"
      ],
      "re_rulebase_set": [
        "core"
      ],
      "regexes_for_supported_peps": [
        "ac[^ ]*",
        "msi[^ ]*",
        "[^ ]*pep_[^ ]*_(pre|post|except|finally)"
      ]
    },
    "shared_memory_instance": "irods_rule_language_rule_engine"
  }
]
$ ils
remote addresses: 172.25.0.3 ERROR: read_client_server_negotiation_message :: received error message -38000
remote addresses: 172.25.0.3 ERROR: _rcConnect: connectToRhost error, server on 9975526fdc81:1247 is probably down status = -38000 SYS_AGENT_INIT_ERR
$ cat /var/log/irods/irods.log | jq '.'
<snip>
{
  "log_category": "legacy",
  "log_facility": "local0",
  "log_level": "error",
  "log_message": "applyRule: -1102000, [-]\t/irods_source/server/re/include/irods/irods_re_plugin.hpp:333:irods::error irods::dynamic_operation_execution_manager<std::tuple<>, RuleExecInfo *, irods::DONT_AUDIT_RULE>::call(const std::string &, const std::string &, OP, As &&...) [T = std::tuple<>, C = RuleExecInfo *, Audit = irods::DONT_AUDIT_RULE, OP = std::function<irods::error (const std::string &, RuleExecInfo *&)>, As = <const std::string &, RuleExecInfo *&>] :  status [NO_MICROSERVICE_FOUND_ERR]  errno [] -- message [default_microservice_manager: no microservice found acChkHostAccessControl]\n\t[-]\t/irods_source/server/re/src/irods_re_plugin.cpp:112:irods::error irods::default_microservice_manager<RuleExecInfo *>::exec_microservice_adapter(std::string, irods::default_ms_ctx, std::list<boost::any> &) :  status [NO_MICROSERVICE_FOUND_ERR]  errno [] -- message [default_microservice_manager: no microservice found acChkHostAccessControl]\n\n",
  "server_host": "9975526fdc81",
  "server_pid": 1981,
  "server_timestamp": "2022-03-08T17:22:50.402",
  "server_type": "agent"
}
<snip>
$ grep acChkHostAccessControl /etc/irods/core.re   
# 15) acChkHostAccessControl - This rule checks the access control by host
# acChkHostAccessControl {msiCheckHostAccessControl; }
```
I'm pretty sure it is required as not every static PEP/policy is implemented in the default core.re. Contrast this with PREP where every static PEP/policy has been implemented in core.py.

Reply:
Oooh, I see.   We just don't have consistency anymore across our rule engine plugins.

So, we can either require the backstop be there (like we do when we also ship an empty `core.py`) - and can reduce the size of the shipped/visible `core.re`... (which I kind of like).

Or, we keep them all internally consistent and they can be fully interchangeable (which I like for different reasons).

Since most static PEPs are going away eventually... trimming down the `core.re` (and depending on the cpp_default) seems... okay?

Reply:
I can confirm that this is happening for me too. Nice catch.
The `LIBRARY_OUTPUT_NAME` property only affects the filename of the built library. The culprit is actually the foreach line: We were treating the contents of the `IRODS_RULE_ENGINE_PLUGINS` list as names of individual lists rather than looking at the `IRODS_RULE_ENGINE_PLUGINS` list itself, so the native rule language REP was the only one even being built. Should be fixed now.

Reply:
Ah, cool. 

And as for the requirement of `default_cpp_policy`... it's only required for a default installation, I should have clarified. Whether or not we want to keep that requirement is perhaps a separate issue. Related to #5952?

Reply:
definitely related to that - yes.   as there are no static PEPs... we can make default `core.x` very small indeed.